packages/controllers/src/snaps/SnapController.ts

```diff
@@ -753,6 +731,11 @@ export class SnapController extends BaseController<
       this._onOutboundResponse,
     );
 
+    this.initializeStateMachine();
+    this.registerMessageHandlers();
+  }
+
+  private initializeStateMachine() {
```

Comment:
Can you add a description for this function? Perhaps stating briefly the need of the state machine to handle snap status state transitions (and side effects in the future.)